Feature/cadastrar cliente

GuilhermeDMueller opened the pull request:
Entidades: Criado parte de informações pessoais do cliente e da construtora;
ClienteRepositorio: Criado Classe repositorio e Interface da Classe.

GuilhermeDMueller committed:
Repositorio.Entidades => Criado parte de informações do cliente

GuilhermeDMueller committed:
Adicionado ClienteRepositorio e IClienteRepositorio.

LucasPerezAlves reviewed (commented):
Colocar específicação detalhada do projeto no commit e não fazer somente um push

GustavoPrim reviewed (commented):
Colocar informações detalhadas do seu projeto

GuilhermeDMueller committed:
Alteração incompleta da tela de cadastro;
Clientes recebe Herança de EntidadeBase;
Implementação da interface e criado corpo de Cadastrar.

Willljdev reviewed (commented):
Nome e NomeConstrutora, não precisa, pois se for pessoa Física terá o campo para por cpf, se for pessoa jurídica terá CNPJ, então o campo nome será apenas um, oque via distinguir se uma pessoa comum ou uma construtora, será a opção escolhida entre CPF ou CNPJ.

Isso vale para os outros pontos em duplicidade

GuilhermeDMueller committed:
Adicionado ClienteContexto e BancoDados, criado ClienteMapeamento e Mapeamentos
Clientes.Nome excluido de entidades,
ClienteContexto e IClienteContexto criados.

GuilhermeDMueller committed:
Cliente.Index Ajustado Subtitulo;
Adicionada pasta Views, e arquivo HTML vazio;
Criada pasta Mapeamentos com arquivo ClienteContextoModelSnapshot;

GustavoPrim commented on Aplicacao.Cliente/Pages/Index.cshtml:
```diff
@@ -5,6 +5,6 @@
-    <p>Learn about <a href="https://docs.microsoft.com/aspnet/core">building Web apps with ASP.NET Core</a>.</p>
-</div>
+    <h1 class="display-4">Cadastramento de Cliente e Construtora</h1>
+    <p>Faça o seu Cadastramanento de Cliente ou Construtora</p>
```
corrigir aqui o erro na frase

GustavoPrim commented on Repositorio/BancoDados/ClienteContexto.cs:
```diff
@@ -0,0 +1,36 @@
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            /* 
```
não precisa colocar esses comentários no trabalho, tem no exemplo pra ser usado corretamente no trabalho

GustavoPrim commented on Repositorio/Mapeamentos/ClienteMapeamento.cs:
```diff
@@ -0,0 +1,18 @@
+        {
+            builder.ToTable("clientes");
+
+            /*builder.HasKey(x => x.Id).HasName("id");
```
não precisa ser exatamente igual ao exemplo, só precisamos usar o necessário

GuilhermeDMueller committed:
Alteração na classe ClieteRepositorio e na interface IClienteRepositorio, sendo criado os metodos Cadastrar e Editar;
Criado classe ClienteService e interface IClienteService, Adicionado metodos de Cadastramento e Edição incompletos;
Criado ClienteCadastrarViewModel, ClienteEditarViewModel e ClienteViewModel;
ClienteViewModel com informações para cadastramento e edição;
ClienteEditarViewModel com Id;
ClienteCadastrarViewModel e ClienteEditarViewModel com Herança de ClienteViewModel;
*Ultimo Pull Request*
Repositorio criado entidade do cliente com as informações necessarias juntmente com Classe ClienteRepositorio e interface
  -IClienteRepositorio;
Alteração incompleta da tela de clientes(continua Incompleta);
Criado ClienteContexto em Banco de dados, clienteMapeamento na pasta Mapeamento;
ClienteIndex Ajustado Subtitulo.

LucasPerezAlves commented on Repositorio/Migrations/ClienteContextoModelSnapshot.cs:
```diff
@@ -0,0 +1,32 @@
+
+                b.HasKey("Id").HasName("id");
+            });
+
```
Retirar linhas em branco

GuilhermeDMueller committed:
Merge branch 'develop' into feature/cadastrar-cliente

GuilhermeDMueller committed:
Retirada lihas em branco na Repositorio/Migrations/ClienteContextoModelSnapshot;
Dados duplicados excluídos em Entidades/Cliente, Repositorios/ClienteRepositorio, Servico/Servicos/ClienteService e ViewModels/ClienteViewModel;

GuilhermeDMueller committed:
Merge branch 'develop' of https://github.com/GustavoPrim/Entra21-TrabalhoFinal-FoxBudgets into feature/cadastrar-cliente
Merge

GuilhermeDMueller committed:
Merge branch 'feature/cadastrar-cliente' of https://github.com/GustavoPrim/Entra21-TrabalhoFinal-FoxBudgets into feature/cadastrar-cliente
Resolvendo merge

GuilhermeDMueller commented:
Últimos 3 Commits.

GustavoPrim committed:
Merge branch 'develop' into feature/cadastrar-cliente

GustavoPrim committed:
Merge branch 'develop' into feature/cadastrar-cliente

LucasPerezAlves commented on Repositorio/BancoDados/ClienteContexto.cs:
```diff
@@ -0,0 +1,36 @@
+            *      dotnet tool install --global dotnet-ef
+            * 1ª etapa - Criar a entidade Raca.cs
+            * 2ª etapa - Criar o mapemanto da entidade para tabela RacaMapeamento.cs
+            * 3ª etapa - Registrar o mapeamento no próprio Contexto
```
Tirar os comentários

GuilhermeDMueller committed:
Merge e adicionado template

GuilhermeDMueller committed:
Merge
